app/models/setting/general.rb

```diff
@@ -29,6 +29,8 @@ def self.default_settings
       self.set('instance_id', N_("Foreman instance ID, uniquely identifies this Foreman instance."), 'uuid', N_('Foreman UUID'), Foreman.uuid),
       self.set('default_locale', N_("Language to use for new users"), nil, N_('Default language'), nil, { :collection => Proc.new { locales } }),
       self.set('default_timezone', N_("Timezone to use for new users"), nil, N_('Default timezone'), nil, { :collection => Proc.new { timezones } }),
+      self.set('enable_facts_storage', N_("Store Fact Values"), false, N_('Store fact values in the database including historical records')),
```

Comment:
IMHO this should not default to `false` because that would be a major change.

Reply:
Well it's not default always. I added migration to check if there are facts / reports, and if yes the default value will be `true`. See latest commit and discussion with Marek.

Reply:
Still, a fresh install will start disabling it in 2.1 with the current implementation. I don't think that's a minor change.

Reply:
"fresh install will start disabling it in 2.1"

Well yeah, that's the point of this PR.

Reply:
And I strongly disagree with that.

Reply:
Well this behavior was there from the beginning, and you didn't say you disagree with this change in your first review, so what happened? @mhulan & @lzap (author of [#28825](https://projects.theforeman.org/issues/28825)) were okay with these changes.

Reply:
And I don't. This is changing the core functionality of the project in a minor release and I think that warrants a larger discussion. For big changes to the core functionality, we usually open [RFCs](https://community.theforeman.org/c/development/rfcs/18) to see if there's agreement.

> you didn't say you disagree with this change in your first review

In my first review (https://github.com/theforeman/foreman/pull/7475#issuecomment-591477315) I said I thought it was a bad idea.

Reply:
To unblock this, I'd personally suggest:

1) let's open RFC for reports and facts separately with a poll having options - a) disable by default, b) enable by default, c) I don't care as long as I can configure that in the installer

2) let's merge this regardless, changing this based on the poll is rather trivial, I don't want to have PR waiting for weeks if we can switch the default value in 5 minutes follow up. If we merge it we can demo it this Thursday and get even more feedback. If we won't, we'll decide based on subset if users who regularly read discourse.

3) from reading the motivation for this change I lean toward having this disabled for reports by default and enabled for facts by default. This should help people who do not realize, they're collecting huge amount of data until it's too late. Facts don't consume that much data. Number of fact values grows with number of hosts and usually is around hundred per source and host. OTOH reports grows faster, for every host, we get the report every 30 minutes and store it IIRC for a month. Usually resource contains a longer string than a value, each reports has many resources.

4) FWIW I don't think this is necessarily a major change. We're not changing any existing setting. We're adding an option to enable/disable something and default to disable on a new installation. It's not a breaking change. However I agree with let's have RFC for what is the best default. We have 2 months to switch this default based on the outcome. As suggested keeping enabled by default wouldn't be considered a major change.

Reply:
>     2\. let's merge this regardless, changing this based on the poll is rather trivial, I don't want to have PR waiting for weeks if we can switch the default value in 5 minutes follow up. If we merge it we can demo it this Thursday and get even more feedback. If we won't, we'll decide based on subset if users who regularly read discourse.

I'd switch this around. Make the default on so it's non-breaking. Then if we want to make it a breaking change, it's a 5 minute change.

>     3\. from reading the motivation for this change I lean toward having this disabled for reports by default and enabled for facts by default. This should help people who do not realize, they're collecting huge amount of data until it's too late. Facts don't consume that much data. Number of fact values grows with number of hosts and usually is around hundred per source and host. OTOH reports grows faster, for every host, we get the report every 30 minutes and store it IIRC for a month. Usually resource contains a longer string than a value, each reports has many resources.

I'd think that reports are tiny unless you're actually using them (i.e., have changed resources in the logs). Facts on the other hand are large regardless of whether you're applying Puppet resources.

Reply:
* Default value changed to true
* For the discussion about default value I will create RFC on discourse

Reply:
> I'd switch this around. Make the default on so it's non-breaking. Then if we want to make it a breaking change, it's a 5 minute change.

I wouldn't call this a breaking change. However I agree with the plan, even though it is not implementing the redmine issue.

Thanks @stejskalleos for quick update, could you please also rephrase the commit message so that it reflects the change? Feel free to just put it here, I can tweak the message on squash.

@ekohl a confirmation from your side would be good to unblock the merge.